drivers/serial/uart_nrfx_uarte.c

```diff
@@ -15,19 +15,28 @@
 #include <nrfx_timer.h>
 #include <zephyr/sys/util.h>
 #include <zephyr/kernel.h>
+#include <zephyr/cache.h>
 #include <soc.h>
 #include <helpers/nrfx_gppi.h>
 #include <zephyr/linker/devicetree_regions.h>
 #include <zephyr/irq.h>
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(uart_nrfx_uarte, CONFIG_UART_LOG_LEVEL);
+#ifdef CONFIG_HAS_NORDIC_DMM
```

Comment:
is this really needed? for legacy platforms these macros in question should evaluate to nothing/`0` : 
- https://github.com/zephyrproject-rtos/zephyr/blob/98e0e65069cf62297c4d03e0ad5b543374f5b735/soc/nordic/common/dmm.h#L61
- https://github.com/zephyrproject-rtos/zephyr/blob/98e0e65069cf62297c4d03e0ad5b543374f5b735/soc/nordic/common/dmm.h#L28

Reply:
removed.